eureka-client/src/main/java/com/netflix/discovery/DiscoveryManager.java

```diff
@@ -40,31 +46,80 @@
  *
  */
 @Deprecated
+@Singleton
 public class DiscoveryManager {
     private static final Logger logger = LoggerFactory.getLogger(DiscoveryManager.class);
-    private DiscoveryClient discoveryClient;
 
-    private EurekaInstanceConfig eurekaInstanceConfig;
-    private EurekaClientConfig eurekaClientConfig;
-    private static final DiscoveryManager s_instance = new DiscoveryManager();
+    private static volatile DiscoveryManager INSTANCE;
+
+    @Inject
+    private volatile Provider<EurekaInstanceConfig> eurekaInstanceConfig = new Provider<EurekaInstanceConfig>() {
```

Comment:
Moving all of these to be providers so that they are still lazy even if DiscoveryManager is eagerly instantiated due to requestStaticInjection